CMakeLists.txt

```diff
@@ -291,9 +291,30 @@ if ( NOT LLVM_TARGET_IS_CROSSCOMPILE_HOST )
     endif ()
 
     check_cxx_compiler_flag("-Werror -Wnested-anon-types" CXX_SUPPORTS_NO_NESTED_ANON_TYPES_FLAG)
-    if( CXX_SUPPORTS_NO_NESTED_ANON_TYPES_FLAG )
+    if (CXX_SUPPORTS_NO_NESTED_ANON_TYPES_FLAG)
       set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-nested-anon-types" )
     endif()
+
+    # remove -Wnon-virtual-dtor for now, since CLR headers are not clean
+    string(REGEX REPLACE "-Wnon-virtual-dtor" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+
+    # if supported, add -Wno-unknown-pragmas, since we use pragma region in LLILC
+    check_cxx_compiler_flag("-Werror -Wno-unknown-pragmas" CXX_SUPPORTS_NO_UNKNOWN_PRAGMAS_FLAG)
+    if (CXX_SUPPORTS_NO_UNKNOWN_PRAGMAS_FLAG)
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unknown-pragmas")
+    endif()
+
+    # if supported, add -Wcovered-switch-default, since we use default NYI's to catch upstream contract changes
```

Comment:
Should this say either 'add -Wno-covered-switch-deafult' or 'remove -Wcovered-switch-default'?